emission/net/api/cfc_webapp.py

```diff
@@ -328,14 +328,18 @@ def summarize_metrics(time_type):
     else:
         old_style = True
         is_return_aggregate = True
+
+    app_config = request.json['app_config'] if 'app_config' in request.json else None
+
```

Comment:
wait, so you are assuming that the client passes in the app config? That sounds fairly bonkers, particularly since the app config is fairly large. We read the app_config directly from the study_config in multiple locations, including in `cfc_webapp` - why would we not just read in the app_config where it is used (presumably in `api/metrics.py`) instead of passing in around?

Reply:
> wait, so you are assuming that the client passes in the app config? That sounds fairly bonkers, particularly since the app config is fairly large.

I agree it's not the best solution. to mitigate the size issue I have the phone sending a partial `app_config` with only `survey_info`, since that is what is really needed for the metrics

> why would we not just read in the app_config where it is used (presumably in `api/metrics.py`) instead of passing in around?

If we did this, then every time we query the server, the server has to query github and wait for the config before proceeding, right? I thought this would probably introduce even more latency.

I also didn't think there was a good way to keep the config in memory because I believe `cfc_webapp` is continuously running and wouldn't receive config updates.

Is there a better way?

Reply:
Ah I see. The standard way to mitigate the performance issue is through caching. You would read the config once a day, and use the cached version otherwise. That would not necessarily introduce a perceptible performance impact.

**However**, that may still leave us with a consistency issue. If the phone config has updated, but the server has not yet updated, I suspect we may have issues because the survey names may be inconsistent across them. The standard way to resolve such consistency issues is to send over the version that the phone is using. The server would cache the last several versions and use the version that the phone sends in.

While thinking through this, I just realized that this might be even more complex. For example, if you had `surveyNameA`, `surveyNameB` and `surveyNameC` in May, and switched to `surveyNameD`, `surveyNameE` and `surveyNameF` in October, you should still apply the older version of the config (e.g. use `surveyNameA`, `surveyNameB` and `surveyNameC` when generating metrics for June in November. That will not work with the current approach of passing the config from the phone either. 

I will merge this for now to unblock ourselves, and we should say that, after a study has been deployed to real users, we will not support changing the surveys. That is a reasonable restriction to impose, given that changing the survey while the data collection is in progress is not a recommended best-practice from a social science perspective - the results are not necessarily comparable with each other.

And then we should figure out the correct performance optimization vs. complexity tradeoff that we want to be at